### ISSUE TITLE
A behavior about ul and li render

### ISSUE THREAD

**Issue by fy0** [markedjs/marked#1281]
**Describe the bug**

```
1. asas

* adasdasd

2. asd45
```

marked & markdown-js:
![image](https://user-images.githubusercontent.com/1579850/40713086-d1765f1a-6431-11e8-9de1-bbe937521995.png)

github & markdown-it & showdown :
![image](https://user-images.githubusercontent.com/1579850/40713226-06b45178-6432-11e8-9b2f-3f9862f6826b.png)

HTML:
```html
<ol>
<li><p>asas</p>
</li>
<li><p>adasdasd</p>
</li>
<li><p>asd45</p>
</li>
</ol>
```

**To Reproduce**

https://marked.js.org/demo/?text=1.%20asas%0A%0A*%20adasdasd%0A%0A2.%20asd45%0A

**Expected behavior**
Actually I don't know which version is right. But github version seems better?

**Comment by UziTech** [comment]
Thank you for reporting this. Looks like a work around would be to add an extra new line between the lines.

```
1. asas


* adasdasd


2. asd45
```

**Comment by fy0** [comment]
@UziTech Oh I got it. But when gfm enable, the situation still happened. Maybe that's the real problem. Thx.

**Comment by UziTech** [comment]
looks like this was fixed in v0.6.0
